Modules/Core/FiniteDifference/include/itkDenseFiniteDifferenceImageFilter.hxx

```diff
@@ -39,15 +41,11 @@ DenseFiniteDifferenceImageFilter<TInputImage, TOutputImage>::CopyInputToOutput()
   }
 
   // Check if we are doing in-place filtering
-  if (this->GetInPlace() && this->CanRunInPlace())
+  if (this->GetInPlace() && this->CanRunInPlace() &&
+      std::equal_to<const void *>{}(input->GetPixelContainer(), output->GetPixelContainer()))
   {
-    const void * const inputPixelContainer = input->GetPixelContainer();
-    const auto * const tempPtr = output.GetPointer();
-    if (tempPtr != nullptr && tempPtr->GetPixelContainer() == inputPixelContainer)
-    {
-      // the input and output container are the same - no need to copy
-      return;
-    }
+    // the input and output container are the same - no need to copy
+    return;
   }
 
   ImageRegionConstIterator<TInputImage> in(input, output->GetRequestedRegion());
```

Comment:
Looks like the nine lines of code that start here could be replaced by a single function call:

    ImageAlgorithm::Copy<TInputImage, TOutputImage>(
      input, output, input->GetRequestedRegion(), output->GetRequestedRegion());

`ImageAlgorithm::Copy` is very much optimized for speed. In fact, I observed that it is even significantly faster, than my personal favorite ITK5 range based iteration, like `std::copy(inputRange.cbegin(), inputRange.cend(), std::begin(outputRange))`.

Reply:
If `ImageAlgorithm::Copy` supports image adaptors, it would be equivalent. I am not sure how important that is in this case.

Reply:
@dzenanz 
> If ImageAlgorithm::Copy supports image adaptors, it would be equivalent.

I think `ImageAlgorithm::Copy` does support image adaptors. But maybe Bradley can confirm... @blowekamp ?

> I am not sure how important that is in this case.

I do believe using  `ImageAlgorithm::Copy` will make `DenseFiniteDifferenceImageFilter::CopyInputToOutput()` really much much faster.  But I don't know how many users actually call `DenseFiniteDifferenceImageFilter::CopyInputToOutput()` on a regular basis. If it isn't being used so much anymore, we might as well leave it alone. What do you think?

Reply:
By the way, is the original line of code OK, saying `in(input, output->GetRequestedRegion())`?  I don't know, but it looks like it should be `in(input, input->GetRequestedRegion())`: `input` instead of `output` 🤔 !

Reply:
We usually limit processing to the requested region, which is set on the output. So it looks right.